### PR TITLE
Change Status to be consistent with Link and Event

### LINKED ISSUE #1037: Remove getters for span status fields from the API
**What are you trying to achieve?**
Remove the following methods from the [Span Status interface](https://github.com/open-telemetry/opentelemetry-specification/blob/dc5b511e5f664e8d98cd317ceceb593cfbbcc14e/specification/trace/api.md#status):
* [GetCanonicalCode](https://github.com/open-telemetry/opentelemetry-specification/blob/dc5b511e5f664e8d98cd317ceceb593cfbbcc14e/specification/trace/api.md#getcanonicalcode)
* [GetDescription](https://github.com/open-telemetry/opentelemetry-specification/blob/dc5b511e5f664e8d98cd317ceceb593cfbbcc14e/specification/trace/api.md#getdescription)

**Additional context.**
The spec is currently inconsistent as it defines the two getters mentioned above on the Span Status object, but no API on the Span itself to get the Status object to call these getters on. In order to make these getters available on the API, implementations either have to expose the enclosed Status via a getter on the span, or have the API designed in a way that allows users to keep a reference to the Status object they created for that span. Note that the spec does not require Status to be immutable.

From #1029, which had the same intent but assumed the getters above would be directly on the span, whereas they are actually on the Status object enclosed by a span:

We do not want any getters on spans, as it leads to inconsistent behavior which we do not want users to rely upon. Due to overhead concerns, we do not want the public API to force the underlying implementation to retain any state. Examples of implementations which do no include: 
* no tracer installed
* sampling enabled
* streaming implementations

Examples of cases where getters could bite users:
* Using status to control application behavior. For example, setting trace status to Error, then switching behavior based on checking the status. This will not always work as expected.

If users which to use this information in their application, they can store it as a separate item in the context object (or as baggage), rather than rely on the span itself.

cc @tedsuo

### LINKED ISSUE #1040: Incomplete sentence in OK status code description
The Span status code `Ok` in the [`StatusCodeCanonical` section](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/api.md#statuscanonicalcode) of the trace API 
has a discription with [a partial sentence](https://github.com/open-telemetry/opentelemetry-specification/pull/966/files#diff-67dcf7b0870e92a2e22456f9e9f52ca9R596)

> The operation has been validated by an Application developers or Operator to have completed successfully, or contain

Possibly "... or contains no error"?

@carlosalberto or @tedsuo do you have an idea of what that was supposed to be?

### PR DESCRIPTION
**Pull request by bogdandrutu**
Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

Fixes #1037
Fixes #1040

### CODE REVIEW

File: specification/trace/api.md [hunk 1]
```diff
@@ -496,16 +491,52 @@ Note that [`RecordException`](#record-exception) is a specialized variant of
 
 #### Set Status
 
-Sets the [`Status`](#status) of the `Span`. If used, this will override the
-default `Span` status, which is `OK`.
+Sets the `Status` of the `Span`. If used, this will override the default `Span`
+status, which is `Unset`.
```
Comment (bogdandrutu) [inline specification/trace/api.md:495]: This is fixing a bug in the specs (I guess)

File: specification/trace/api.md [hunk 2]
```diff
@@ -496,16 +491,52 @@ Note that [`RecordException`](#record-exception) is a specialized variant of
...
+- `Error`
+  - The operation contains an error.
 
 The Span interface MUST provide:
 
-- An API to set the `Status` where the new status is the only argument. This
-  SHOULD be called `SetStatus`.
+- An API to set the `Status` where the new `Status` properties are passed as
+  arguments. This SHOULD be called `SetStatus`. This API takes the
+  `StatusCanonicalCode`, and optional description.
```
Comment (yurishkuro) [inline specification/trace/api.md:516]:
```suggestion
- An API to set the `Status`. This SHOULD be called `SetStatus`.
  This API takes the `StatusCanonicalCode`, and optional `Description`, 
  either as individual parameters or as an immutable object encapsulating them, 
  whichever is most appropriate for the language.
```

File: specification/trace/api.md [hunk 3]
```diff
@@ -496,16 +491,52 @@ Note that [`RecordException`](#record-exception) is a specialized variant of
...
-default `Span` status, which is `OK`.
+Sets the `Status` of the `Span`. If used, this will override the default `Span`
+status, which is `Unset`.
 
-Only the value of the last call will be recorded, and implementations are free
-to ignore previous calls.
+`Status` is semantically defined by the following properties:
+
+- `StatusCanonicalCode` represents the canonical set of `Status` codes.
+- Description represents the descriptive message of the `Status`.
```
Comment (yurishkuro) [inline specification/trace/api.md:500]:
```suggestion
- `StatusCanonicalCode` that represents the canonical set of `Status` codes.
- Optional `Description` that provides a descriptive message of the `Status`.
```

File: specification/trace/api.md [hunk 4]
```diff
@@ -496,16 +491,52 @@ Note that [`RecordException`](#record-exception) is a specialized variant of
...
+Application developers and Operators may set the status code to `Ok`.
+
+Analysis tools SHOULD respond to an `Ok` status by suppressing any errors they
+would otherwise generate. For example, to suppress noisy errors such as 404s.
+
+Only the value of the last call will be recorded, and implementations are free
+to ignore previous calls.
+
+Note that there are multiple ways to implement this API, such as: overloading,
+variadic arguments, or define an immutable object to be passed to the API calls.
```
Comment (yurishkuro) [inline specification/trace/api.md:539]: not necessary with the edit to the API description above.

File: specification/trace/api.md [hunk 5]
```diff
@@ -496,16 +491,52 @@ Note that [`RecordException`](#record-exception) is a specialized variant of
...
-Sets the [`Status`](#status) of the `Span`. If used, this will override the
-default `Span` status, which is `OK`.
+Sets the `Status` of the `Span`. If used, this will override the default `Span`
+status, which is `Unset`.
 
-Only the value of the last call will be recorded, and implementations are free
-to ignore previous calls.
+`Status` is semantically defined by the following properties:
+
+- `StatusCanonicalCode` represents the canonical set of `Status` codes.
```
Comment (yurishkuro) [inline specification/trace/api.md:499]: Why are we using such a long name as StatusCanonicalCode? Are there non-canonical status codes?

File: specification/trace/api.md (hunk 4 above)
Comment (bogdandrutu) [reply specification/trace/api.md:540]: Correct, removed

File: specification/trace/api.md [hunk 6]
```diff
@@ -496,16 +491,52 @@ Note that [`RecordException`](#record-exception) is a specialized variant of
...
+- Description represents the descriptive message of the `Status`.
+
+`StatusCanonicalCode` has the following values:
+
+- `Unset`
+  - The default status.
+- `Ok`
+  - The operation has been validated by an Application developers or Operator to
+    have completed successfully, or contain
```
[the remark this answers is not in the document]
Comment (bogdandrutu) [reply specification/trace/api.md:508]: Paying the debt of https://github.com/open-telemetry/opentelemetry-specification/issues/1040

File: specification/trace/api.md [hunk 7]
```diff
@@ -496,16 +491,52 @@ Note that [`RecordException`](#record-exception) is a specialized variant of
...
+publish their own conventions, including status codes.
+
+Generally, Instrumentation Libraries SHOULD NOT set the status code to `Ok`,
+unless explicitly configured to do so. Instrumention libraries SHOULD leave the
+status code as `Unset` unless there is an error, as described above.
+
+Application developers and Operators may set the status code to `Ok`.
+
+Analysis tools SHOULD respond to an `Ok` status by suppressing any errors they
+would otherwise generate. For example, to suppress noisy errors such as 404s.
```
[the remark this answers is not in the document]
Comment (bogdandrutu) [reply specification/trace/api.md:534]: Good question, I was not changing this text, will file the issue https://github.com/open-telemetry/opentelemetry-specification/issues/1068 to track this.

File: specification/trace/api.md (hunk 5 above)
Comment (bogdandrutu) [reply specification/trace/api.md]: Good question, I was not changing this, will file the issue #1068 to track this.

**Review by tigrannajaryan — APPROVED** [review]
LGTM, from what I see it is primarily editorial changes and content reshulfling.

**Review by carlosalberto — APPROVED** [review]

**Review by iNikem — APPROVED** [review]

**Review by anuraaga — APPROVED** [review]

File: specification/trace/api.md [hunk 8]
```diff
@@ -496,16 +491,50 @@ Note that [`RecordException`](#record-exception) is a specialized variant of
...
+
+Only the value of the last call will be recorded, and implementations are free
+to ignore previous calls.
```
Comment (arminru) [inline specification/trace/api.md:537]: This is only taken over but could be refined to be worded consistently (MUST/SHOULD/MAY) either here or in a follow-up.

**Review by arminru — APPROVED** [review]

File: specification/trace/api.md [hunk 9]
```diff
@@ -496,16 +491,50 @@ Note that [`RecordException`](#record-exception) is a specialized variant of
...
+
+- `StatusCanonicalCode` that represents the canonical set of `Status` codes.
+- Optional `Description` that provides a descriptive message of the `Status`.
+
+`StatusCanonicalCode` has the following values:
+
+- `Unset`
+  - The default status.
+- `Ok`
+  - The operation has been validated by an Application developers or Operator to
```
Comment (arminru) [inline specification/trace/api.md:507]:
```suggestion
  - The operation has been validated by an Application developer or Operator to
```

**Pull request merged**